### PR TITLE
:sparkles: - feat: implemented iconinitials and added a tooltip option

### PR DESCRIPTION
**Pull request by Xaohs**
allows merging of https://github.com/maykinmedia/open-archiefbeheer/pull/269

### CODE REVIEW

File: src/components/iconinitials/iconinitials.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  initials,
+  size = "2em",
+  fontSize,
+  ...props
+}) => {
+  const calculatedFontSize = fontSize ? fontSize : `calc(${size} * 0.5)`;
+  return (
+    <div
+      className={"mykn-iconinitials"}
+      style={{
```
Comment (svenvandescheur) [inline src/components/iconinitials/iconinitials.tsx:27]: Can we 

- Use a a (container) tag P for (font) styling?
- Don't use inline styling, we can probably use the em unit for sizing? If we need a bigger size we can make a size prop.

File: src/components/iconinitials/iconinitials.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+};
+
+/**
+ * IconInitials component. Used to display an icon to show initial letters.
+ * @param initials
+ * @param props
+ * @constructor
+ */
+export const Iconinitials: React.FC<IconInitialsProps> = ({
+  initials,
```
Comment (svenvandescheur) [inline src/components/iconinitials/iconinitials.tsx:18]: Can we:

 - Use the full name as input string
 - Generate the initials 
 - Use the full name on a aria-label attribute

File: src/components/tooltip/tooltip.stories.tsx
Comment (svenvandescheur) [inline src/components/tooltip/tooltip.stories.tsx:1]: We we're going to use the dropdown right, so this isn't needed anymore?

File: src/style/tokens/base.scss [hunk 3]
```diff
@@ -149,6 +149,7 @@
   --page-color-secondary-contrast: var(--typography-color-body);
   --page-color-shadow: rgba(12, 12, 12, 0.3);
   --page-container-size: 1240px;
+  --page-color-background-alt2: #e3dbff;
```
Comment (svenvandescheur) [inline src/style/tokens/base.scss:152]: I think we don't need a n additional token here, as we can construct a similar color by altering the HSL values found in:

```
  --brand-color-hue: #{hue($brand-color)};
  --brand-color-saturisation: #{saturation($brand-color)};
  --brand-color-lightness: #{lightness($brand-color)};
  ```

**Review by svenvandescheur — CHANGES_REQUESTED** [review]
Nice approach making this a reusable component, a few remarks but I like to get this in :) Oh: I think you mistyped: Iconinitials (should be IconInitials?)

File: src/components/iconinitials/iconinitials.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  fontSize?: string;
+};
+
+/**
+ * IconInitials component. Used to display an icon to show initial letters.
+ * @param initials
+ * @param props
+ * @constructor
+ */
+export const Iconinitials: React.FC<IconInitialsProps> = ({
```
Comment (svenvandescheur) [inline src/components/iconinitials/iconinitials.tsx:17]: IconInitials

File: src/components/iconinitials/iconinitials.tsx (hunk 2 above)
Comment (Xaohs) [reply src/components/iconinitials/iconinitials.tsx:18]: I like this idea, we let the component generate the initials for us. And indeed aria!!!

File: src/components/iconinitials/iconinitials.tsx (hunk 1 above)
Comment (Xaohs) [reply src/components/iconinitials/iconinitials.tsx:27]: Good idea

File: src/components/iconinitials/iconinitials.scss [hunk 5]
```diff
@@ -0,0 +1,44 @@
+.mykn-iconinitials {
+  --brand-color-hue: 253deg;
```
Comment (svenvandescheur) [inline src/components/iconinitials/iconinitials.scss:2]: We need to alter the current primary color, not redefine it. We want to:

 - Respect the current configured color (may be overridden).
 - Alter it to our needs.


Having access to the HSL (hue satursation ligthness)  values allow the for instance create a lighter variant. of the color. But we may need to do som calc.

If dynamic values don't work we need te try blending with a white layer (:before/:after)

File: src/components/iconinitials/iconinitials.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+    // Get the first letter of the last name (or last word in the name)
+    const lastInitial = nameParts[nameParts.length - 1].charAt(0).toUpperCase();
+
+    // Combine the initials
+    return firstInitial + lastInitial;
+  };
+
+  return (
+    <P
```
Comment (svenvandescheur) [inline src/components/iconinitials/iconinitials.tsx:40]: By overriding the classnamee, ther is no .P classname, therefore the P styling is not there. So this needs to restructured a bit.

File: src/components/iconinitials/iconinitials.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+}) => {
+  const getInitials = (name: string): string => {
+    // Split the name into an array of words
+    const nameParts = name.trim().split(" ");
+
+    // Get the first letter of the first name
+    const firstInitial = nameParts[0].charAt(0).toUpperCase();
+
+    // Get the first letter of the last name (or last word in the name)
+    const lastInitial = nameParts[nameParts.length - 1].charAt(0).toUpperCase();
```
Comment (svenvandescheur) [inline src/components/iconinitials/iconinitials.tsx:33]: This worked surprisingly well :)

**Review by svenvandescheur — CHANGES_REQUESTED** [review]

**Review by svenvandescheur — APPROVED** [review]

**Pull request merged**